hedera-node/hedera-mono-service/src/main/java/com/hedera/services/fees/calculation/schedule/txns/ScheduleCreateResourceUsage.java

```diff
@@ -47,25 +53,64 @@ public boolean applicableTo(TransactionBody txn) {
     @Override
     public FeeData usageGiven(TransactionBody txn, SigValueObj svo, StateView view)
             throws InvalidTxBodyException {
-        var op = txn.getScheduleCreate();
         var sigUsage =
                 new SigUsage(
                         svo.getTotalSigCount(), svo.getSignatureSize(), svo.getPayerAcctSigCount());
 
-        final long lifetimeSecs;
+        lifetimeSecs = calculateLifeTimeSecs(txn);
+        final long defaultLifeTimeSecs = dynamicProperties.scheduledTxExpiryTimeSecs();
+        return scheduleOpsUsage.scheduleCreateUsage(
+                txn, sigUsage, lifetimeSecs, defaultLifeTimeSecs);
+    }
 
-        if (op.hasExpirationTime() && dynamicProperties.schedulingLongTermEnabled()) {
-            lifetimeSecs =
-                    Math.max(
-                            0L,
-                            op.getExpirationTime().getSeconds()
-                                    - txn.getTransactionID()
-                                            .getTransactionValidStart()
-                                            .getSeconds());
-        } else {
-            lifetimeSecs = dynamicProperties.scheduledTxExpiryTimeSecs();
+    @Override
+    public boolean hasSecondaryFees() {
+        return true;
+    }
+
+    /**
+     * Since long term scheduled transactions can be scheduled far from future, adds a fee based on
+     * number of minutes until scheduled time and size of schedule transaction. This fee is added
+     * only to the transactions scheduled beyond the defaultLifeTimeSecs.
+     *
+     * <p>Fee calculation is as follows : If lifetimeSecs &lt;= defaultLifeTimeSecs, no additional
+     * fee is charged If lifetimeSecs &gt; defaultLifeTimeSecs, an additional price is charged based
+     * on serialized size of the scheduled transaction and the expiration tiem is calculated.
+     *
+     * <p>For example if the defaultLifeTimeSecs is {@code 1800}, priceIncrement is {@code 20000000}
+     * tinycents, and secondaryFeeBytesPerMonth is {@code 128} bytes, then for a scheduled
+     * transaction that is scheduled to execute anytime after 1800 secs, an extra fee of $0.002 for
+     * 128 bytes/month is charged.
+     *
+     * @param txn schedule create transaction body
+     * @return fee object for additional fee
+     */
+    @Override
+    public FeeObject secondaryFeesFor(final TransactionBody txn) {
+        final long secondaryFeeTinyCents = dynamicProperties.scheduleTxSecondaryFee();
+        final int secondaryFeeBPM = dynamicProperties.scheduleTxSecondaryFeeBytesPerMonth();
+        final long defaultLifeTimeSecs = dynamicProperties.scheduledTxExpiryTimeSecs();
+
+        if (lifetimeSecs > defaultLifeTimeSecs) {
+            final var serializedSize =
+                    txn.getScheduleCreate().getScheduledTransactionBody().getSerializedSize();
+            final var numerator =
```

Comment:
🤔 In the [issue](https://github.com/hashgraph/hedera-services/issues/4154), we have
```
  Price = $0.01
    + (minutesTillScheduledTime > 30  ? 1 : 0)
    * $0.002
    * Math.Ceiling(sizeOfScheduledTransaction / 128)
    * Math.Ceiling(minutesUntilScheduledTime / (30*24*60))
```
This makes me think the `nonDegenerateDiv()` may be in the wrong place here.

I would have expected,
```
final var numBpmChunks = ESTIMATOR_UTILS.nonDegenerateDiv(serializedSize, secondaryFeeBPM);
final var numMonths = ESTIMATOR_UTILS.nonDegenerateDiv(lifetimeSecs, ONE_MONTH_IN_SECS);
final var additionalServiceFee = clampedMultiply(
    secondaryFeeTinyCents, clampedMultiply(numBpmChunks, numMonths));
    
```

Reply:
I did that because any transaction with size < 128 bytes should be charged less, and also transaction with life time > 1800 secs and < 1 month also should be charged less.

```
final var numBpmChunks = ESTIMATOR_UTILS.nonDegenerateDiv(serializedSize, secondaryFeeBPM);
final var numMonths = ESTIMATOR_UTILS.nonDegenerateDiv(lifetimeSecs, ONE_MONTH_IN_SECS);
```
These will return 1 if the value is less than 1 right ?